app/master/slave.py

```diff
@@ -236,6 +238,14 @@ def _expected_session_header(self):
 
         return headers
 
+    def set_heartbeat(self, time):
+        self._heartbeat = time
+
+    def is_responsive(self, time, heartbeat_frequency):
```

Comment:
1. Can we rename `time` to `current_time` or `now` (that's both clearer, and it avoids shadowing the time module, which is a common python module)
2. Rename self._heartbeat to self._last_checkin
3. rename heartbeat_frequency to heartbeat_period
4. This method can be rewritten as
```python
return (current_time - self._last_checkin) < heartbeat_period
```
5. Given that this method is effectively a one-liner, it might be cleaner to just expose self._last_checkin to the caller, and let the caller execute this one line itself
(the argument against this point is that the name is_responsive describes what this one line does better than the inlined code does. I'm not sure if I agree with that or not. If you want to make that argument i'll give it some thought and see if I agree)